src/libasr/pass/python_bind.cpp

```diff
@@ -0,0 +1,459 @@
+    body.push_back(al, ASRUtils::STMT(ASRUtils::make_SubroutineCall_t_util(al, f.base.base.loc, sym_Py_DecRef, nullptr,
+                        args_Py_DecRef3.p, args_Py_DecRef3.n, nullptr, nullptr, false, false)));
+
+    // reassignment
+    f.m_body = body.p;
+    f.n_body = body.n;
+    ASRUtils::get_FunctionType(f)->m_abi = ASR::abiType::Source;
+    ASRUtils::get_FunctionType(f)->m_deftype = ASR::deftypeType::Implementation;
+}
+
+void pass_python_bind(Allocator &al, ASR::TranslationUnit_t &unit, const PassOptions &pass_options) {
+    if (pass_options.c_skip_bindpy_pass) {
+        // FIXME: C backend supports arrays, it is used in bindpy_02 to bindpy_04 tests.
+        //  This pass currently does not handle any aggregate types.
+        //  Once we include support for aggregate types, we should remove this check, and
+        //  remove the `c_backend` variable from PassOptions.
+        //  We will also need to remove the special handling of BindPython ABI in C
+        //  backend as it would be handled by this ASR pass.
+        return;
+    }
+
+    std::vector<ASRUtils::ASRFunc> fns;
+    fns.push_back({"Py_Initialize", {}, ASRUtils::VOID});
+    fns.push_back({"Py_IsInitialized", {}, ASRUtils::I32});
+    // fns.push_back({"PyRun_SimpleString", {STR}, ASRUtils::I32});
+    fns.push_back({"Py_DecodeLocale", {ASRUtils::STR, ASRUtils::PTR}, ASRUtils::PTR});
+    fns.push_back({"PySys_SetArgv", {ASRUtils::I32, ASRUtils::PTR_TO_PTR}, ASRUtils::VOID});
+    fns.push_back({"Py_FinalizeEx", {}, ASRUtils::I32});
+    fns.push_back({"PyUnicode_FromString", {ASRUtils::STR}, ASRUtils::PTR});
+    fns.push_back({"PyUnicode_AsUTF8AndSize", {ASRUtils::PTR, ASRUtils::PTR}, ASRUtils::STR});
+    fns.push_back({"PyImport_Import", {ASRUtils::PTR}, ASRUtils::PTR});
+    fns.push_back({"Py_DecRef", {ASRUtils::PTR}, ASRUtils::VOID});
+    fns.push_back({"Py_IncRef", {ASRUtils::PTR}, ASRUtils::VOID});
+    fns.push_back({"PyObject_GetAttrString", {ASRUtils::PTR, ASRUtils::STR}, ASRUtils::PTR});
+    fns.push_back({"PyTuple_New", {ASRUtils::I32}, ASRUtils::PTR});
+    fns.push_back({"PyTuple_SetItem", {ASRUtils::PTR, ASRUtils::I32, ASRUtils::PTR}, ASRUtils::I32});
+    fns.push_back({"PyObject_CallObject", {ASRUtils::PTR, ASRUtils::PTR}, ASRUtils::PTR});
+    fns.push_back({"PyLong_AsLongLong", {ASRUtils::PTR}, ASRUtils::I64});
+    fns.push_back({"PyLong_AsUnsignedLongLong", {ASRUtils::PTR}, ASRUtils::U64});
+    fns.push_back({"PyLong_FromLongLong", {ASRUtils::I64}, ASRUtils::PTR});
+    fns.push_back({"PyLong_FromUnsignedLongLong", {ASRUtils::U64}, ASRUtils::PTR});
+    fns.push_back({"PyFloat_FromDouble", {ASRUtils::F64}, ASRUtils::PTR});
+    fns.push_back({"PyFloat_AsDouble", {ASRUtils::PTR}, ASRUtils::F64});
+    fns.push_back({"PyBool_FromLong", {ASRUtils::I32}, ASRUtils::PTR});
+    fns.push_back({"PyObject_IsTrue", {ASRUtils::PTR}, ASRUtils::I32});
+
+    bool included_cpython_funcs = false;
```

Comment:
I think `included_cpython_funcs` is not needed. We can simply call ` ASRUtils::declare_functions(al, fns, *l, unit.m_symtab, "Python.h")` here. We skip the whole `python_bind` pass if `--enable-cpython` is not provided.


Also suggested here https://github.com/lcompilers/lpython/pull/2796#discussion_r1704413641.

Reply:
The `--enable-cpython` flag sets `CompilerOptions.enable_cpython` to true. We do not have access to `CompilerOptions` within an ASR pass. We only have access to `PassOptions`.
Should I move `enable_cpython` to `PassOptions` then? Or have it in both places in `CompilerOptions` and `PassOptions`, so that we don't break any other code.

Reply:
We can move it to pass options.

Reply:
I have made the changes. Hope I understood it correctly and this is what you wanted.